src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,196 @@
+                + "WHERE cars.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            return getCarList(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get a list of cars from database.", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET model = ?, manufacturer_id = ?  WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement
+                         = connection.prepareStatement(query)) {
+            statement.setString(FIRST_INDEX, car.getModel());
+            statement.setLong(SECOND_INDEX, car.getManufacturer().getId());
+            statement.setLong(THIRD_INDEX, car.getId());
+            statement.executeUpdate();
+            removeDriverFromCar(car);
+            addDrivers(car.getDriverList(), car.getId());
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car: " + car, e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE cars SET is_deleted = TRUE WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(FIRST_INDEX, id);
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete car by id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.id, manufacturer_id, model, m.id, name, country "
+                + "FROM cars c JOIN cars_drivers ON driver_id = id "
+                + "JOIN manufacturers m ON c.id = m.id "
+                + "WHERE c.is_deleted = FALSE AND driver_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(FIRST_INDEX, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            return getCarList(resultSet);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get a list of cars", e);
+        }
+    }
+
+    private void addDrivers(List<Driver> driverList, Long carId) {
+        String query = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query,
+                         Statement.RETURN_GENERATED_KEYS)) {
```

Comment:
You don't use statement.getGeneratedKeys() in this method
```suggestion
                 PreparedStatement statement = connection.prepareStatement(query)) {
```